### DISCUSSION TITLE
地址簿下双击连接一台远端pc后地址簿的pc列表都显示离线状态，过一会儿又刷新成在线

### DISCUSSION THREAD

**Discussion by MemoryClear** [lejianwen/rustdesk-api#86] [General]
如题，不确定是客户端的问题还是api服务器在连接远端pc时将所有pc状态置为离线状态了。客户端修改关联标签的时候同样也是先置离线状态，再重新刷新状态

**Comment by lejianwen** [comment]
Api 服务不能改变机器在线状态

**Reply by MemoryClear** [reply]
不是改变实际状态，是提交修改之后，绿点在线状态会短暂变成灰色，然后几秒后恢复绿色，时间间隔不固定。客户端版本是最新的1.3.6

**Reply by lejianwen** [reply]
如果不使用API Server呢？也有这样的情况吗？
我这边未能复现
看是否是这个原因? https://github.com/lejianwen/rustdesk-api/issues/50#issuecomment-2457277675

按照逻辑来讲，api server是不会影响到客户端上`在线/离线`的颜色`绿色/橙色`

**Reply by MemoryClear** [reply]
> 如果不使用API Server呢？也有这样的情况吗？ 我这边未能复现 看是否是这个原因? [#50 (comment)](https://github.com/lejianwen/rustdesk-api/issues/50#issuecomment-2457277675)
> 
> 按照逻辑来讲，api server是不会影响到客户端上`在线/离线`的颜色`绿色/橙色`

我使用[kingmo888](https://github.com/kingmo888)的api服务没有这个问题。

**Reply by lejianwen** [reply]
好，我大概明白问题在哪了，明天修复下，感谢反馈

**Reply by MemoryClear** [reply]
> 好，我大概明白问题在哪了，明天修复下，感谢反馈

辛苦大佬

**Reply by lejianwen** [reply]
我刚看了一下，不使用个人版就没这个问题
修改选项`RUSTDESK_API_RUSTDESK_PERSONAL=0`，
或者配置文件中
```
rustdesk:
  personal: 0
```

~个人版这个问题可能无法修复，我将返回的`online`去掉也还是会这样，我明天再看看客户端的代码再试试~


我想到了，明天修复

**Reply by lejianwen** [reply]
我仔细看了，无法修复，api返回的`online`不会影响到客户端中`peer`的`online`。
这个只能等官方更新了，但是这个不是开源版的功能，只能pro版提issue估计。


如果比较介意这个问题，可以不用是个人版。

**Reply by MemoryClear** [reply]
谢谢大佬，已关闭个人版api。

**Reply by lejianwen** [reply]
不过 关闭个人版 后，要注意下， 后台添加地址簿名称就没用了，每人都只有一个默认的地址簿

**Reply by MemoryClear** [reply]
> 不过 关闭个人版 后，要注意下， 后台添加地址簿名称就没用了，每人都只有一个默认的地址簿

好的大佬。

**Reply by MemoryClear** [reply]
> 不过 关闭个人版 后，要注意下， 后台添加地址簿名称就没用了，每人都只有一个默认的地址簿

大佬，能否加一个控制RUSTDESK_API_RUSTDESK_PERSONAL=0的时候禁用地址簿名称的新增功能（前端隐藏地址簿管理菜单或者相关接口控制都可以），查询通讯录设备列表的接口只查询默认地址簿的设备（目前好像是会查询返回所有地址簿的设备）

**Reply by lejianwen** [reply]
> 禁用地址簿名称的新增功能（前端隐藏地址簿管理菜单或者相关接口控制都可以）

这个没必要，因为还有很多地方有多地址簿的功能


> 查询通讯录设备列表的接口只查询默认地址簿的设备

这个可以我更新下


一般来说还是建议启用`RUSTDESK_API_RUSTDESK_PERSONAL`
`Legacy`模式的接口就是整个一大坨数据提交

**Reply by MemoryClear** [reply]
> > 禁用地址簿名称的新增功能（前端隐藏地址簿管理菜单或者相关接口控制都可以）
> 
> 这个没必要，因为还有很多地方有多地址簿的功能
> 
> > 查询通讯录设备列表的接口只查询默认地址簿的设备
> 
> 这个可以我更新下
> 
> 一般来说还是建议启用`RUSTDESK_API_RUSTDESK_PERSONAL` `Legacy`模式的接口就是整个一大坨数据提交

感谢大佬，辛苦大佬了

**Reply by lejianwen** [reply]
已经更新了，你试试
